src/components/download/DownloadDialog.js

```diff
@@ -0,0 +1,169 @@
+        hasName: PropTypes.bool.isRequired,
+        hasLegend: PropTypes.bool.isRequired,
+        toggleDownloadDialog: PropTypes.func.isRequired,
+        toggleDownloadShowName: PropTypes.func.isRequired,
+        toggleDownloadShowLegend: PropTypes.func.isRequired,
+        setDownloadLegendPosition: PropTypes.func.isRequired,
+        classes: PropTypes.object.isRequired,
+    };
+
+    state = {
+        error: null,
+    };
+
+    render() {
+        const {
+            showDialog,
+            hasName,
+            showName,
+            hasLegend,
+            showLegend,
+            legendPosition,
+            toggleDownloadShowName,
+            toggleDownloadShowLegend,
+            setDownloadLegendPosition,
+            classes,
+        } = this.props;
+
+        if (!showDialog) {
+            return null;
+        }
+
+        const isSupported = downloadSupport();
+
+        return (
+            <Dialog open={showDialog} onClose={this.onClose}>
+                <DialogTitle disableTypography={true}>
+                    {i18n.t('Download map')}
+                </DialogTitle>
+                <DialogContent className={classes.content}>
+                    {isSupported && !this.state.error ? (
+                        <Fragment>
+                            <div className={classes.checkbox}>
+                                <Checkbox
+                                    label={i18n.t('Show name')}
+                                    checked={showName}
+                                    disabled={!hasName}
+                                    onCheck={toggleDownloadShowName}
+                                />
+                            </div>
+                            <div className={classes.checkbox}>
+                                <Checkbox
+                                    label={i18n.t('Show legend')}
+                                    checked={showLegend}
+                                    disabled={!hasLegend}
+                                    onCheck={toggleDownloadShowLegend}
+                                />
+                            </div>
+                            {hasLegend &&
+                                showLegend && (
+                                    <LegendPosition
+                                        position={legendPosition}
+                                        onChange={setDownloadLegendPosition}
+                                    />
+                                )}
+                        </Fragment>
+                    ) : (
+                        i18n.t('Map download is not supported by your browser. Try Google Chrome or Firefox.')
+                    )}
+                </DialogContent>
+                <DialogActions>
+                    <Button color="primary" onClick={this.onClose}>
+                        {isSupported ? i18n.t('Cancel') : i18n.t('Close')}
+                    </Button>
+                    {isSupported &&
```

Comment:
As written I think it could be possible for an error to occur, displaying 'Map download is not supported by your browser. Try Google Chrome or Firefox' but still showing an enabled 'Download' button.  It probably wouldn't enter that state in the wild very often, but we should probably use the same conditional (`isSupported && !this.state.error`) everywhere (line 116 and 118 here)

Reply:
Good catch! Added: 
const isSupported = downloadSupport() && !this.state.error;